python/ray/dag/compiled_dag_node.py

```diff
@@ -1033,45 +1033,12 @@ def _get_or_compile(
                         output_to_readers[task].append(downstream_task)
                 fn = task.dag_node._get_remote_method("__ray_call__")
                 for output, readers in output_to_readers.items():
-                    dag_nodes = [reader.dag_node for reader in readers]
-                    read_by_multi_output_node = False
-                    for dag_node in dag_nodes:
-                        if isinstance(dag_node, MultiOutputNode):
-                            read_by_multi_output_node = True
-                            break
-
+                    read_by_multi_output_node = any(
+                        isinstance(reader.dag_node, MultiOutputNode)
+                        for reader in readers
+                    )
                     reader_and_node_list: List[Tuple["ray.actor.ActorHandle", str]] = []
                     if read_by_multi_output_node:
-                        if len(readers) != 1:
-                            raise ValueError(
-                                "DAG outputs currently can only be read by the "
-                                "driver or the same actor that is also the "
-                                "InputNode, not by both the driver and actors.",
-                            )
-
-                        # This node is a multi-output node, which means it will
-                        # only be read by the driver or the actor that is also
-                        # the InputNode.
-
-                        # TODO(jhumphri): Handle case where there is an actor,
-                        # other than just the driver actor, also reading the
-                        # output from the `task` node. For example, the following
-                        # currently does not work:
-                        # def test_blah(ray_start_regular):
-                        #     a = Actor.remote(0)
-                        #     b = Actor.remote(10)
-                        #     with InputNode() as inp:
-                        #         x = a.inc.bind(inp)
-                        #         y = b.inc.bind(x)
-                        #         dag = MultiOutputNode([x, y])
-
-                        #     compiled_dag = dag.experimental_compile()
-                        #     output_channel = compiled_dag.execute(1)
-                        #     result = output_channel.read()
-                        #     print(result)
-
-                        #     compiled_dag.teardown()
-
                         assert self._proxy_actor is not None
                         for reader in readers:
                             reader_and_node_list.append(
```

Comment:
@ruisearch42 so this seems to work by just removing the readers != 1 check, a little confused why, if the proxy_actor is just getting added to the list for every reader even non-drivers how does it work? the tests seem to pass but i feel like there's something else that could be going wrong